contrib/storage-hive/core/src/main/java/org/apache/drill/exec/store/hive/HiveUtilities.java

```diff
@@ -398,17 +399,13 @@ public static void addConfToJob(final JobConf job, final Properties properties)
    * Wrapper around {@link MetaStoreUtils#getPartitionMetadata(Partition, Table)} which also adds parameters from table
    * to properties returned by {@link MetaStoreUtils#getPartitionMetadata(Partition, Table)}.
    *
-   * @param partition {@link Partition} instance
-   * @param table {@link Table} instance
+   * @param partition the source of partition level parameters
+   * @param table     the source of table level parameters
    * @return properties
    */
-  public static Properties getPartitionMetadata(final Partition partition, final Table table) {
+  public static Properties getPartitionMetadata(final HivePartition partition, final HiveTable table) {
     final Properties properties;
-    // exactly the same column lists for partitions and table
-    // stored only in table to reduce physical plan serialization
-    if (partition.getSd().getCols() == null) {
-      partition.getSd().setCols(table.getSd().getCols());
-    }
+    restoreColumns(table, partition);
```

Comment:
could this restoreColumns fail the purpose of this fix? Since it is setting columns back to each partition, if getPartitionMetadata was called before the final physical plan generated, then these columns could be printed out to the plan again, could you check this possibility?